cpp/src/neighbors/detail/cagra/cagra_search.cuh

```diff
@@ -140,6 +142,61 @@ void search_main(raft::resources const& res,
                  raft::device_matrix_view<DistanceT, int64_t, raft::row_major> distances,
                  CagraSampleFilterT sample_filter = CagraSampleFilterT())
 {
+  if constexpr (!std::is_same_v<CagraSampleFilterT,
```

Comment:
Can you pull this out into a separate function that can be invoked here please? This `search_main` function is gettig pretty massive.

Reply:
Done